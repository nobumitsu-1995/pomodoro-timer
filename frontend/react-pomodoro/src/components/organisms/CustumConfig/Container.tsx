import React, { useEffect, useState } from 'react'
import { useDispatch } from 'react-redux'
import { tokenGetSelector } from 'src/feature/selectors/token'
import { custumConfigsSelector } from 'src/feature/selectors/custumConfig'
import { updateCustumConfig } from 'src/feature/slices/custumConfig'
import { setCustumTimerConfig } from 'src/feature/slices/timerConfig'
import { useSelector } from 'src/feature/store'
import { api } from 'src/lib/functions/axios'
import {
  validateCycle,
  validateCycleToLongRestTime,
  validateTimerConfig,
} from 'src/lib/functions/validation'
import Presenter from './Presenter'

const Container: React.FC = () => {
  const dispatch = useDispatch()
  const token = useSelector(tokenGetSelector)
  const custumConfigs = useSelector(custumConfigsSelector)
  const [selectNum, setSelectNum] = useState(0)
  const [custumConfig, setCustumConfig] = useState(custumConfigs[0])
  const [errors, setErrors] = useState({
    workTime: '',
    restTime: '',
    cycle: '',
    longRestTime: '',
    cycleToLongRestTime: '',
  })

  useEffect(() => {
    setCustumConfig(custumConfigs[selectNum])
    dispatch(setCustumTimerConfig(custumConfigs[selectNum]))
  }, [selectNum, custumConfigs])

  const formItems = [
    {
      id: 'workTime',
      label: 'work time',
      value: custumConfig.workTime,
      error: errors.workTime || '',
    },
    {
      id: 'restTime',
      label: 'rest time',
      value: custumConfig.restTime,
      error: errors.restTime || '',
    },
    {
      id: 'cycle',
      label: 'cycle',
      value: custumConfig.cycle,
      error: errors.cycle || '',
    },
    {
      id: 'longRestTime',
      label: 'long rest time',
      value: custumConfig.longRestTime,
      error: errors.longRestTime || '',
    },
    {
      id: 'cycleToLongRestTime',
      label: (
        <>
          Cycles to <br /> long rest periods
        </>
      ),
      value: custumConfig.cycleToLongRestTime,
      error: errors.cycleToLongRestTime || '',
    },
  ]

  const changeEdit = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target
    validateForm(name, Number(value))
    setCustumConfig((prev) => ({
      ...prev,
      [name]: value,
    }))
  }

  const validateForm = (name: string, value: number) => {
    switch (name) {
      case 'cycle':
        setErrors((prev) => ({
          ...prev,
          [name]: validateCycle(value),
        }))
        break
      case 'cycleToLongRestTime':
        setErrors((prev) => ({
          ...prev,
          [name]: validateCycleToLongRestTime(value),
        }))
        break
      case 'workTime':
      case 'restTime':
      case 'longRestTime':
        setErrors((prev) => ({
          ...prev,
          [name]: validateTimerConfig(value),
        }))
        break
      default:
        break
    }
  }

  const clickUpdate = () => {
    if (checkFormsValidaty()) return

    api(token)
      .patch(`/v1/custum_config/${custumConfig._id}/update`, custumConfig)
      .then((res) => {
        dispatch(updateCustumConfig(res.data))
        dispatch(setCustumTimerConfig(res.data))
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const checkFormsValidaty = () => {
    Object.entries(custumConfig).forEach((data) => {
      validateForm(data[0], Number(data[1]))
    })

    return (
      Object.values(errors).filter((value) => {
        return value !== ''
      }).length !== 0
    )
  }

  const onChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const { value } = event.target
    setSelectNum(parseInt(value))
  }

  return (
    <Presenter
      formItems={formItems}
      configLength={5}
      clickUpdate={clickUpdate}
      changeEdit={changeEdit}
      onChangeSelect={onChangeSelect}
    />
  )
}

export default Container
