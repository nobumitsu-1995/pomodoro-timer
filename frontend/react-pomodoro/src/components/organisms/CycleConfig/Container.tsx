import React, { useCallback, useEffect, useState } from 'react'
import { useDispatch } from 'react-redux'
import { cycleSelector } from '../../../feature/selectors/timerConfig'
import { useSelector } from '../../../feature/store'
import { updateCycle as reduxUpdateCycle } from '../../../feature/slices/timerConfig'
import Presenter from './Presenter'
import { validateCycle } from 'src/lib/functions/validation'

const Container: React.FC = () => {
  const dispatch = useDispatch()
  const updateCycle = useCallback(
    (cycle: number) => {
      dispatch(reduxUpdateCycle(cycle))
    },
    [dispatch]
  )
  const globalCycle = useSelector(cycleSelector)
  const [cycle, setCycle] = useState(globalCycle)
  const [error, setError] = useState('')

  useEffect(() => {
    setCycle(globalCycle)
  }, [globalCycle])

  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    const _cycle = Number(e.target.value)
    setError(validateCycle(_cycle))
    setCycle(Number(_cycle))
  }

  const handleClickButton = () => {
    const _error = validateCycle(cycle)
    setError(_error)
    if (_error) return
    updateCycle(cycle)
  }

  const handleOnClickUp = () => {
    const _cycle = cycle + 1
    const _error = validateCycle(_cycle)
    setError(_error)

    if (_error) return
    setCycle(_cycle)
    updateCycle(_cycle)
  }

  const handleOnClickDown = () => {
    const _cycle = cycle - 1
    const _error = validateCycle(_cycle)
    setError(_error)

    if (_error) return
    setCycle(_cycle)
    updateCycle(_cycle)
  }

  return (
    <Presenter
      value={cycle}
      error={error}
      onChange={handleChangeInput}
      onClick={handleClickButton}
      onClickUp={handleOnClickUp}
      onClickDown={handleOnClickDown}
    />
  )
}

export default Container
