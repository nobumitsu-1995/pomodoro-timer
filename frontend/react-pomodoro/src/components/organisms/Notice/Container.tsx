import React, { useState } from 'react'
import { noticesSelector } from '../../../feature/selectors/notice'
import { useSelector } from '../../../feature/store'
import { NoticeType } from '../../../lib/types/modelType'
import Presenter from './Presenter'

const Container: React.FC = () => {
  const notices = useSelector(noticesSelector)
  const [currentNotice, setCurrentNotice] = useState<NoticeType | undefined>()

  return (
    <Presenter
      notices={notices}
      notice={currentNotice}
      setCurrentNotice={setCurrentNotice}
    />
  )
}

export default Container
