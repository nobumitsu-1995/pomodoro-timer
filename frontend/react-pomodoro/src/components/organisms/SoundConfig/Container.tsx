import React, { useCallback } from 'react'
import useSound from 'use-sound'
import RestFinish from '../../../assets/sounds/RestFinish.mp3'
import { useDispatch } from 'react-redux'
import { volumeSelector } from '../../../feature/selectors/soundConfig'
import {
  updateVolume as reduxUpdateVolume,
  muteVolume as reduxMuteVolume,
  unMuteVolume as reduxUnMuteVolume,
} from '../../../feature/slices/soundConfig'
import { useSelector } from '../../../feature/store'
import Presenter from './Presenter'

const Container: React.FC = () => {
  const volume = useSelector(volumeSelector)
  const globalVolume = useSelector(volumeSelector)
  const [playRestFinish] = useSound(RestFinish, { volume: volume / 100 })
  const dispatch = useDispatch()
  const updateVolume = useCallback(
    (_volume: number) => {
      dispatch(reduxUpdateVolume(_volume))
    },
    [dispatch]
  )
  const muteVolume = useCallback(() => {
    dispatch(reduxMuteVolume())
  }, [dispatch])
  const unMuteVolume = useCallback(() => {
    dispatch(reduxUnMuteVolume())
  }, [dispatch])

  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    const _volume = Number(e.target.value)
    updateVolume(_volume)
  }

  const handleClickButton = () => {
    if (globalVolume === 0) {
      unMuteVolume()
    } else {
      muteVolume()
    }
  }

  return (
    <Presenter
      isMuted={globalVolume === 0}
      onClick={handleClickButton}
      onChange={handleChangeInput}
      value={globalVolume}
      onClickTry={playRestFinish}
    />
  )
}

export default Container
