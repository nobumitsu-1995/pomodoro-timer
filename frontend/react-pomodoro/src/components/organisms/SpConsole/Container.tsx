import { useAuth0 } from '@auth0/auth0-react'
import React, { useState } from 'react'
import Console from 'src/components/templates/Console'
import { Notice } from '..'
import UserAchievement from '../UserAchievement/Container'
import Presenter from './Presenter'

const Container = () => {
  const [isOpen, setIsOpen] = useState(false)
  const [content, setContent] = useState<string>('')
  const { isAuthenticated } = useAuth0()

  const touchButton = (id: string) => {
    if (id === content) {
      return setIsOpen(!isOpen)
    }
    if (isOpen) {
      return setContent(id)
    }
    setContent(id)
    setIsOpen(!isOpen)
  }

  const CONTENT_LIST = [
    {
      id: 'notice',
      content: <Notice />,
    },
    {
      id: 'help',
      content: <Notice />,
    },
    {
      id: 'config',
      content: <Console />,
    },
    {
      id: 'user',
      content: <UserAchievement />,
    },
  ]

  return (
    <Presenter
      isLoggedIn={isAuthenticated}
      isOpen={isOpen}
      content={CONTENT_LIST.find((v) => v.id === content)?.content}
      touchButton={touchButton}
    />
  )
}

export default Container
