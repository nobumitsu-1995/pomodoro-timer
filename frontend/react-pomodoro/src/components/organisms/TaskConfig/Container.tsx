import React, { useState } from 'react'
import { TaskType } from 'src/lib/types/modelType'
import { useDispatch } from 'react-redux'
import { useSelector } from 'src/feature/store'
import { currentTaskSelector, tasksSelector } from 'src/feature/selectors/task'
import {
  addTask,
  setCurrentTask,
  deleteTask,
  updateTask,
} from 'src/feature/slices/tasks'
import { api } from '../../../lib/functions/axios'
import Presenter from './Presenter'
import { tokenGetSelector } from 'src/feature/selectors/token'

const Container: React.FC = () => {
  const dispatch = useDispatch()
  const currentTask = useSelector(currentTaskSelector)
  const token = useSelector(tokenGetSelector)
  const tasks = useSelector(tasksSelector)

  const [task, setTask] = useState({
    title: '',
    id: '',
  })
  const [isEditMode, setIsEditMode] = useState(false)

  const onClickTask = (task: TaskType) => {
    dispatch(setCurrentTask(task))
  }

  const onClickDeleteTask = (id: string) => {
    if (!confirm('削除してよろしいですか？')) return
    api(token)
      .delete(`/v1/task/${id}/delete`)
      .then(() => {
        dispatch(deleteTask(id))
        setTask({ title: '', id: '' })
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const onClickEditTask = (task: TaskType) => {
    setIsEditMode(true)
    setTask({ title: task.title, id: task._id })
  }

  const onClickUpdateTask = () => {
    if (task.title.length === 0) return

    api(token)
      .patch(`/v1/task/${task.id}/update`, {
        title: task.title,
      })
      .then((res) => {
        dispatch(updateTask(res.data))
        dispatch(setCurrentTask(res.data))
        setTask({ title: '', id: '' })
        setIsEditMode(false)
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const onClickCreateTask = () => {
    if (task.title.length === 0) return

    api(token)
      .post('/v1/task/create', {
        title: task.title,
      })
      .then((res) => {
        dispatch(addTask(res.data))
        dispatch(setCurrentTask(res.data))
        setTask({ title: '', id: '' })
      })
      .catch((e) => {
        console.error(e)
      })
  }

  const onChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target
    setTask({
      ...task,
      title: value,
    })
  }

  return (
    <Presenter
      task={task}
      tasks={tasks}
      currentTask={currentTask}
      isEditMode={isEditMode}
      onClickTask={onClickTask}
      onClickDeleteTask={onClickDeleteTask}
      onClickEditTask={onClickEditTask}
      onClickUpdateTask={onClickUpdateTask}
      onClickCreateTask={onClickCreateTask}
      onChangeInput={onChangeInput}
    />
  )
}

export default Container
