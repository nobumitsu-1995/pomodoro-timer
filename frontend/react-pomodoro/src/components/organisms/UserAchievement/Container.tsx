import React, { Fragment } from 'react'
import { tokenGetSelector } from 'src/feature/selectors/token'
import { useSelector } from 'src/feature/store'
import useUserAchievement from 'src/lib/hooks/useUserAchievement'
import Presenter from './Presenter'

const UserAchievement = () => {
  const token = useSelector(tokenGetSelector)
  const { totalAchievements, isLoading } = useUserAchievement({ token })

  const labels = totalAchievements.map(
    (achievement) => achievement.taskId.title
  )

  const datas = totalAchievements.map((achievement) => achievement.time)

  return (
    <Fragment>
      <Presenter
        labels={labels}
        datas={datas}
        achievements={totalAchievements}
        isLoading={isLoading}
      />
    </Fragment>
  )
}

export default UserAchievement
