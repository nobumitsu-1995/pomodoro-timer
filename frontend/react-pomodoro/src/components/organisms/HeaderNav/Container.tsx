import React from 'react'
import { useAuth0 } from '@auth0/auth0-react'
import Presenter from './Presenter'
import Notice from '../Notice/Container'
import UserAchievement from '../UserAchievement/Container'

const Container: React.FC = () => {
  const { isAuthenticated } = useAuth0()

  const listButtons = [
    {
      modalContent: <UserAchievement />,
      icon: 'person',
      isHide: !isAuthenticated,
      desc: 'UserInfo',
    },
    {
      modalContent: <Notice />,
      icon: 'notifications',
      desc: 'Notifications',
    },
    {
      modalContent: <p>Help Page</p>,
      icon: 'help',
      desc: 'Help',
    },
  ]

  return <Presenter listButtons={listButtons} isLoggedIn={isAuthenticated} />
}

export default Container
