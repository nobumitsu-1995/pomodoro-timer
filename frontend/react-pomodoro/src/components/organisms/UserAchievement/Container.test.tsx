/* eslint-disable react/react-in-jsx-scope */
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import Theme from 'src/assets/styles/Theme'
import { store } from 'src/feature/store'
import Container from './Container'

describe('TimeConfig', () => {
  test('snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Theme>
          <Container />
        </Theme>
      </Provider>
    ).asFragment()
    expect(tree).toMatchSnapshot()
  })
})
