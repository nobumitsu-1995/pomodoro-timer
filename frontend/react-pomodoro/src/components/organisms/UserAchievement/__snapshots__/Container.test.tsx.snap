// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeConfig snapshot 1`] = `
<DocumentFragment>
  loading
</DocumentFragment>
`;
